python/src/linalg.cpp

```diff
@@ -210,4 +210,28 @@ void init_linalg(py::module_& parent_module) {
             array([[-2.23607, -3.57771],
                    [0, 0.447214]], dtype=float32)
       )pbdoc");
+  m.def(
+      "svd",
+      &svd,
+      "a"_a,
+      py::kw_only(),
+      "stream"_a = none,
+      R"pbdoc(
+        svd(a: array, *, stream: Union[None, Stream, Device] = None) -> (array, array, array)
+
+        The Singular Value Decomposition (SVD) of the input matrix.
+
+        This function supports arrays with at least 2 dimensions. When the input
+        has more than two dimensions, the function operates in "stacked mode" like
+        numpy.
+
+        Args:
+            a (array): Input array.
+            stream (Stream, optional): Stream or device. Defaults to ``None``
+              in which case the default stream of the default device is used.
+
+        Returns:
+            tuple(array, array, array): The ``U``, ``S``, and ``Vt`` matrices, such that
```

Comment:
I notice you call it `Vt` instead of `Vh` as in the numpy docs. Do you know what the `t` (or `h`) is intended to stand for? Just curious..

Reply:
Both lapack and NumPy return the last term in the decomposition (`V`) as its transpose. To be precise, it's not just the transpose, but the hermitian conjugate (or conjugate transpose, see https://en.wikipedia.org/wiki/Conjugate_transpose). The difference only matters for complex numbers, as for real numbers transpose and conjugate transpose are identical.

Since we only support float32 for now, I've used `t` to avoid confusion (and match lapack signatures).